tests/test_client.py

```diff
@@ -44,17 +49,19 @@ def kant_ocrdzip(ocrd_identifier):
     yield dest
     unlink(dest)
 
+
 def test_login(client):
-    skip("temporary")
     assert client.token is None
     client.login()
     assert client.token
 
+
 def test_post(client, kant_ocrdzip, ocrd_identifier):
     client.login()
-    # client.post('bar', ocrd_identifier)
-    print(client.post(kant_ocrdzip))
-    assert 0
+    pid = client.post(kant_ocrdzip)
+    print(f"Uploaded ocrd-zip: {pid}")
```

Comment:
Here you would put assertions on caplog.records or caplog.record_tuples.

Pytest will show output from loggers in case of failure, but not from print.

Reply:
@joschrew https://github.com/bertsky/nmalign/blob/master/tests/test_all.py for an example of what @bertsky means.

Reply:
I simply would want to know here if uploading fails or not (assertion does that) and in case of success get the PID of the newly created archive to possibly delete it after the test. Now I set the `-s` flag in the Makefile when running pytests which works for this

Reply:
`pytest -s` is almost always a bad idea. In a normal (successful) run, you don't want to be molested by all the other information on the stdandard streams. You want a neat summary. But `-s` is not only indiscriminate, it also prevents checking for output explicitly (via `capfd` or others).

If you want the PID regardless of the pytest results, consider writing to a file and then `cat`ing that file afterwards (if you do it in a CI, you can even write it to the summary or report shown on the website).